packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -174,6 +179,126 @@ class Plugin {
     );
   }
 
+  /// Queries the `pubpsec` and returns the `platforms` sub map.
+  ///
+  /// Throws ToolExit if cannot find the `platforms:` map.
+  static YamlMap getPlatformsYamlMap(YamlMap pubspec) {
+    if (pubspec == null) {
+       throwToolExit('Invalid pubspec.yaml', exitCode: 2);
+    }
+    final YamlMap flutterConfig = pubspec['flutter'] as YamlMap;
+    if (flutterConfig == null) {
+      throwToolExit('Did not find valid `flutter` map in pubspec.yaml', exitCode: 2);
+    }
+    final YamlMap pluginConfig = flutterConfig['plugin'] as YamlMap;
+    if (pluginConfig == null) {
+      throwToolExit('Did not find valid `plugin` map in pubspec.yaml', exitCode: 2);
+    }
+    if (pluginConfig['platforms'] == null || pluginConfig['platforms'] is! YamlMap) {
+      throwToolExit(_invalidPlatformsErrorMessage, exitCode: 2);
+    }
+
+    return pluginConfig['platforms'] as YamlMap;
+  }
+
+  /// Updates the pubspec.yaml under `projectDir`, adding each platform in `platforms` under `plugin: platforms:`
+  ///
+  /// If the dummy platform `some_platform` exists, remove it.
+  /// For example, if the original `plugin` map is:
+  /// ```
+  ///   flutter:
+  ///      plugin:
+  ///        platforms:
+  ///          some_platform:
+  ///            pluginClass: somePluginClass
+  /// ```
+  /// and the `platforms` is specified as `['ios','android'], the yaml map becomes:
+  /// ```
+  ///    flutter:
+  ///      plugin:
+  ///        platforms:
+  ///          android:
+  ///            package: <androidIdentifier>
+  ///            pluginClass: <pluginClass>
+  ///          ios:
+  ///            pluginClass: <pluginClass>
+  /// ```
+  ///
+  /// This method doesn't support the legacy plugin pubspec format and will throw error if try to update a pubspec.yaml that uses the legacy format.
+  static Future<void> updatePubspecWithPlatforms(String projectDir, List<String> platforms, String pluginClass, String androidIdentifier) async {
+    if (platforms.isEmpty) {
+      return;
+    }
+
+    final String pubspecPath = globals.fs.path.join(projectDir, 'pubspec.yaml');
+    if (!globals.fs.file(pubspecPath).existsSync()) {
+      return;
+    }
+    final YamlMap pubspec = loadYaml(globals.fs.file(pubspecPath).readAsStringSync()) as YamlMap;
+    final List<String> errors = validatePluginYaml(pubspec);
+    if (errors.isNotEmpty) {
+      throwToolExit('Invalid plugin specification: \n${errors.join('\n')}');
+    }
+    try {
+      // The format of the updated pubspec might not be preserved.
+      final List<String> platformsToAdd = List<String>.from(platforms);
+      final YamlMap platformsMap = getPlatformsYamlMap(pubspec);
+      final List<String> existingPlatforms = platformsMap.keys.cast<String>().toList();
+      platformsToAdd.removeWhere((String platform) => existingPlatforms.contains(platform));
+      if (platformsToAdd.isEmpty) {
+        return;
+      }
+      final File pubspecFile = globals.fs.file(pubspecPath);
```

Comment:
What if instead of trying to edit their pubspec and potentially corrupting it or changing their formatting or accidentally removing comments, we print out something like (assuming `--platforms=web,macos`):

```
In order to support web and macos, update your pubspec.yaml platforms map.
See https://flutter.dev/docs/development/packages-and-plugins/developing-packages#plugin-platforms for details.

  plugin:
    platforms:
      android:
        package: com.example.hello (actually put the existing map we parsed for them)
        pluginClass: HelloPlugin (actually put the existing map we parsed for them)
      ios:
        pluginClass: HelloPlugin (actually put the existing map we parsed for them)
      macos:
        pluginClass: HelloPlugin (put the suggested pluginClass)
      web:
        pluginClass: HelloPlugin (put the suggested pluginClass)
        fileName: hello_web.dart (put the default fileName)
```

Reply:
Given the time constraints and the downside of getting this wrong, this seems like a reasonable compromise in the short term. We can always circle back later if the feedback is negative